common/fs.cpp

```diff
@@ -152,6 +152,18 @@ WriteStream *FSNode::createWriteStream() const {
 	return _realNode->createWriteStream();
 }
 
+bool FSNode::createDirectory() const {
+	if (_realNode == nullptr)
+		return false;
+
+	if (_realNode->exists()) {
+		return false;
+	}
+
```

Comment:
I would check ``if (!_realNode->isDirectory())`` and do a warning like above in ``createReadStream`` and ``createWriteStream`` but with logic reversed.
Additionnaly I would check here ``if (!_realNode->isWritable())`` and warn to be sure we can create the directory. I am not sure about this one as it may already be checked in ``_realNode->create()``
Maybe this kind of check should be added to ``createReadStream`` and ``createWriteStream`` too.

Reply:
Thanks for pointing this! Added some warnings at aea4c2e14f0be2c01be861c59f34c22f8e8fbd08.

As for `isWritable`, it's semantics are useful for existing files/directories only. It is implemented as `access(_path.c_str(), W_OK) == 0` for POSIX backend, `_access(_path.c_str(), W_OK) == 0` for Windows, etc... So it is unrelated to ability to create a directory and current AbstractFSNode does not provide any information on such permissions.

Reply:
Hmm, sorry. I didn't understood well this part of code. I thought ``_realNode`` was the parent directory into which we created a new directory.
As ``_realNode`` is directory to create my comment was totally absurd.
Sorry again for the noise. :smile: